### PR TITLE
#3004: require Authentication Token in Require Auth

### LINKED ISSUE #3004: Clear token button broken on settings page
Steps to reproduce
- In the extension options go to the Settings screen
- Hit "Clear Token"

Expected Behavior
- Token is cleared from local storage and page reloads
- Extension shows the 2-step onboarding screen

Actual behavior
- Page reloads
- User still shows as logged in
- When you visit the Admin Console, it shows the "succesfully linked extension" message indicating that the auth token might be getting cleared, but the user profile state isn't?

### PR DESCRIPTION
**Pull request by twschiller**
What does this PR do?
---
- Closes #3004 
- Requires the Authentication header token in RequireAuth

### CODE REVIEW

File: src/options/pages/SetupPage.tsx [hunk 1]
```diff
@@ -126,13 +117,6 @@ const SetupPage: React.FunctionComponent = () => {
                   account
                 </Button>
               </div>
-
-              <div className="mt-2">
-                <small className="text-muted">
```
Comment (twschiller) [inline src/options/pages/SetupPage.tsx:131]: Removing because this is not currently suppored

Comment (fregante) [reply src/options/pages/SetupPage.tsx:131]: This closes https://github.com/pixiebrix/pixiebrix-extension/issues/835 💯

File: src/auth/RequireAuth.tsx [hunk 2]
```diff
@@ -45,10 +59,16 @@ const RequireAuth: React.FC<RequireAuthProps> = ({
...
+    []
+  );
+  const { isLoading: meLoading, error, data: me } = useGetMeQuery();
+
+  const isLoading = tokenLoading || meLoading;
 
   useEffect(() => {
-    // Before we get the very first response from API, do nothing, use the cached version.
+    // Before we get the very first response from API, do nothing, use the AuthRootState persisted with redux-persist.
     if (isLoading) {
```
[the remark this answers is not in the document]
Comment (BALEHOK) [reply src/auth/RequireAuth.tsx:72]: Otherwise the logic inside `setAuth` function should be updated to take `hasToken` into account.

**Review by BALEHOK — CHANGES_REQUESTED** [review]

File: src/auth/RequireAuth.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (twschiller) [reply src/auth/RequireAuth.tsx:72]: You're correct that should be meLoading

**Review by BALEHOK — APPROVED** [review]

**Pull request merged**